cardinal/search/search.go

```diff
@@ -24,30 +24,95 @@ type cache struct {
 // So it is not recommended to create a new search every time you want
 // to filter entities with the same search.
 type Search struct {
-	archMatches *cache
-	filter      filter.ComponentFilter
-	namespace   string
-	reader      gamestate.Reader
-	wCtx        engine.Context
-	filterFuncs []func(id types.EntityID) bool
+	archMatches             *cache
+	filter                  filter.ComponentFilter
+	namespace               string
+	reader                  gamestate.Reader
+	wCtx                    engine.Context
+	componentPropertyFilter PredicateEvaluator
 }
 
 // NewSearch creates a new search.
 // It receives arbitrary filters that are used to filter entities.
-func NewSearch(wCtx engine.Context, filter filter.ComponentFilter) *Search {
+func NewSearch(wCtx engine.Context) *Search {
+	return NewSearchWithFilter(wCtx, nil)
+}
+
+func NewSearchWithFilter(wCtx engine.Context, componentFilter filter.ComponentFilter) *Search {
+	return &Search{
+		archMatches:             &cache{},
+		filter:                  componentFilter,
+		namespace:               wCtx.Namespace(),
+		reader:                  wCtx.StoreReader(),
+		wCtx:                    wCtx,
+		componentPropertyFilter: nil,
+	}
+}
+
+func (s *Search) Contains(component ...componentWrapper) *Search {
+	if s.filter != nil {
+		panic("Search already contains a filter.")
+	}
```

Comment:
is this case ever possible? would it be bad if we just let it continue and overwrite the filter if there was one?

Reply:
ahh i see. thanks!